### PR TITLE
Document it is possible to use Java Records for immutable configuration with @ConfigurationProperties

### LINKED ISSUE #10086: Document it is possible to use Java Records for immutable configuration with @ConfigurationProperties
### Issue description

We have an example in a the [Micronaut Data with Java Records](https://guides.micronaut.io/latest/micronaut-java-records-gradle-java.html#immutable-configuration-with-java-records) guide. 

However, in the Micronaut Core reference documentation we don't have any example of immutable configuration using Java records. 

> [From a performance perspective records are better than interfaces](https://github.com/micronaut-projects/micronaut-views/pull/613#discussion_r1386450561)

### PR DESCRIPTION
**Pull request by wetted**
closes #10086

### CODE REVIEW

File: src/main/docs/guide/config/immutableConfig.adoc [hunk 1]
```diff
@@ -55,6 +55,26 @@ snippet::io.micronaut.docs.config.immutable.VehicleSpec[tags="start",indent=0,ti
 
 The above example prints: `"Ford Engine Starting V8 [rodLength=7B.0]"`
 
+==== Using Java Record Classes to define immutable configuration
+
+For Java language applications, it's also possible to use https://docs.oracle.com/en/java/javase/17/language/records.html[Java Record Classes] for immutable configuration with ann:context.annotation.ConfigurationProperties[]. For example:
+
+[source, java]
+----
+import io.micronaut.context.annotation.ConfigurationProperties;
```
Comment (sdelamo) [inline src/main/docs/guide/config/immutableConfig.adoc]: this should be in code not in asciidoctor and we should have a test for the code shown in the documentation.

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by wetted** [comment]
Sergio I am unable to get the generated HTML to render the Java Record snippet for some reason. Investigating...

**Review by sdelamo — APPROVED** [review]

**Comment by timyates** [comment]
@wetted running `./gradlew docs -x htmlSanityCheck` locally, I see

image

Is that not what you see?

**Comment by wetted** [comment]
> @wetted running `./gradlew docs -x htmlSanityCheck` locally, I see
> 
> image
> Is that not what you see?

No. Probably a caching issue on my mac. I'll check it again

**Comment by wetted** [comment]
@timyates thanks for checking. It is a cache issue with Chrome on my Mac. I see it just fine with Safari and Firefox, so I'm good with that.

**Review by graemerocher — APPROVED** [review]

**Pull request merged**